From cee1178eeeb6fb9e5c115ae45f460997c120e631 Mon Sep 17 00:00:00 2001
From: bilux <i.bilux@gmail.com>
Date: Sat, 21 Mar 2020 09:46:06 +0100
Subject: [PATCH] add mtk ril requests & unsol responses

Signed-off-by: bilux <i.bilux@gmail.com>
---
 .../com/android/internal/telephony/RIL.java   | 53 +++++++++++++++++++
 1 file changed, 53 insertions(+)

diff --git a/src/java/com/android/internal/telephony/RIL.java b/src/java/com/android/internal/telephony/RIL.java
index 7352dc556..d8c0a627b 100644
--- a/src/java/com/android/internal/telephony/RIL.java
+++ b/src/java/com/android/internal/telephony/RIL.java
@@ -133,6 +133,33 @@
     static final boolean RILJ_LOGV = false; // STOPSHIP if true
     static final int RIL_HISTOGRAM_BUCKET_COUNT = 5;
 
+    static final int REFRESH_SESSION_RESET = 6; /* Session reset */
+
+    static final int RIL_REQUEST_VENDOR_BASE = 2000;
+    static final int RIL_REQUEST_MODEM_POWEROFF = (RIL_REQUEST_VENDOR_BASE + 10);
+    //static final int RIL_REQUEST_DUAL_SIM_MODE_SWITCH  = (RIL_REQUEST_VENDOR_BASE + 11);
+    //static final int RIL_REQUEST_USIM_AUTHENTICATION  = (RIL_REQUEST_VENDOR_BASE + 27);
+    static final int RIL_REQUEST_MODEM_POWERON = (RIL_REQUEST_VENDOR_BASE + 28);
+    static final int RIL_REQUEST_RESUME_REGISTRATION  = (RIL_REQUEST_VENDOR_BASE + 65);
+    //static final int RIL_REQUEST_SIM_INTERFACE_SWITCH  = (RIL_REQUEST_VENDOR_BASE + 68);
+    static final int RIL_REQUEST_SET_CALL_INDICATION = (RIL_REQUEST_VENDOR_BASE + 86);
+    static final int RIL_REQUEST_EMERGENCY_DIAL = (RIL_REQUEST_VENDOR_BASE + 87);
+    static final int RIL_REQUEST_SET_ECC_SERVICE_CATEGORY = (RIL_REQUEST_VENDOR_BASE + 88);
+    static final int RIL_REQUEST_SET_ECC_LIST = (RIL_REQUEST_VENDOR_BASE + 89);
+    static final int RIL_REQUEST_GENERAL_SIM_AUTH = (RIL_REQUEST_VENDOR_BASE + 90);
+    static final int RIL_REQUEST_QUERY_AVAILABLE_NETWORK_WITH_ACT = (RIL_REQUEST_VENDOR_BASE + 95);
+    static final int RIL_REQUEST_SWITCH_CARD_TYPE = (RIL_REQUEST_VENDOR_BASE + 131);
+
+    static final int RIL_UNSOL_VENDOR_BASE = 3000;
+    static final int RIL_UNSOL_RESPONSE_PS_NETWORK_STATE_CHANGED = (RIL_UNSOL_VENDOR_BASE + 15);
+    static final int RIL_UNSOL_RESPONSE_REGISTRATION_SUSPENDED = (RIL_UNSOL_VENDOR_BASE + 24);
+    static final int RIL_UNSOL_INCOMING_CALL_INDICATION = (RIL_UNSOL_VENDOR_BASE + 42);
+    static final int RIL_UNSOL_CALL_INFO_INDICATION = (RIL_UNSOL_VENDOR_BASE + 49);
+    static final int RIL_UNSOL_MD_STATE_CHANGE = (RIL_UNSOL_VENDOR_BASE + 53);
+    static final int RIL_UNSOL_SET_ATTACH_APN = (RIL_UNSOL_VENDOR_BASE + 73);
+    //static final int RIL_UNSOL_MAL_AT_INFO = (RIL_UNSOL_VENDOR_BASE + 74);
+    //static final int RIL_UNSOL_MAIN_SIM_INFO = (RIL_UNSOL_VENDOR_BASE + 75);
+
     /**
      * Wake lock timeout should be longer than the longest timeout in
      * the vendor ril.
@@ -4962,6 +4989,20 @@ static String requestToString(int request) {
                 return "RIL_REQUEST_SET_SIGNAL_STRENGTH_REPORTING_CRITERIA";
             case RIL_REQUEST_SET_LINK_CAPACITY_REPORTING_CRITERIA:
                 return "RIL_REQUEST_SET_LINK_CAPACITY_REPORTING_CRITERIA";
+            case RIL_REQUEST_RESUME_REGISTRATION:
+                return "RIL_REQUEST_RESUME_REGISTRATION";
+            case RIL_REQUEST_SET_CALL_INDICATION:
+                return "RIL_REQUEST_SET_CALL_INDICATION";
+            case RIL_REQUEST_EMERGENCY_DIAL:
+                return "RIL_REQUEST_EMERGENCY_DIAL";
+            case RIL_REQUEST_SET_ECC_SERVICE_CATEGORY:
+                return "RIL_REQUEST_SET_ECC_SERVICE_CATEGORY";
+            case RIL_REQUEST_SET_ECC_LIST:
+                return "RIL_REQUEST_SET_ECC_LIST";
+            case RIL_REQUEST_MODEM_POWEROFF:
+                return "RIL_REQUEST_MODEM_POWEROFF";
+            case RIL_REQUEST_MODEM_POWERON:
+                return "RIL_REQUEST_MODEM_POWERON";
             default: return "<unknown request>";
         }
     }
@@ -5072,6 +5113,18 @@ static String responseToString(int request) {
                 return "RIL_UNSOL_KEEPALIVE_STATUS";
             case RIL_UNSOL_PHYSICAL_CHANNEL_CONFIG:
                 return "RIL_UNSOL_PHYSICAL_CHANNEL_CONFIG";
+            case RIL_UNSOL_RESPONSE_REGISTRATION_SUSPENDED:
+                return "RIL_UNSOL_RESPONSE_REGISTRATION_SUSPENDED";
+            case RIL_UNSOL_INCOMING_CALL_INDICATION:
+                return "RIL_UNSOL_INCOMING_CALL_INDICATION";
+            case RIL_UNSOL_CALL_INFO_INDICATION:
+                return "RIL_UNSOL_CALL_PROGRESS";
+            case RIL_UNSOL_MD_STATE_CHANGE:
+                return "RIL_UNSOL_MD_STATE_CHANGE";
+            case RIL_UNSOL_SET_ATTACH_APN:
+                return "RIL_UNSOL_SET_ATTACH_APN";
+            case RIL_UNSOL_RESPONSE_PS_NETWORK_STATE_CHANGED:
+                return "RIL_UNSOL_RESPONSE_PS_NETWORK_STATE_CHANGED";
             default:
                 return "<unknown response>";
         }
