From 28302d40b8f8749657be050e2547ed2e2461c8a7 Mon Sep 17 00:00:00 2001
From: Bilux <i.bilux@gmail.com>
Date: Thu, 30 Jan 2020 21:28:01 +0100
Subject: [PATCH] whitelist /proc/ged from zygote

---
 core/jni/fd_utils.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/core/jni/fd_utils.cpp b/core/jni/fd_utils.cpp
index 92a542676bcd..a94d49ae3e58 100644
--- a/core/jni/fd_utils.cpp
+++ b/core/jni/fd_utils.cpp
@@ -43,6 +43,7 @@ static const char* kPathWhitelist[] = {
   "/dev/ion",
   "/dev/dri/renderD129", // Fixes b/31172436
   "/system/framework/org.lineageos.platform-res.apk",
+  "/proc/ged", //TRONX2100 for MTK_HARDWARE
 };
 
 static const char kFdPath[] = "/proc/self/fd";
